cli/mender-artifact/write.go

```diff
@@ -119,6 +119,24 @@ func writeRootfs(c *cli.Context) error {
 			return cli.NewExitError("SSH error: "+err.Error(), 1)
 		}
 		defer os.Remove(rootfsFilename)
+
+		// run fsck
+		fstype, err := imgFilesystemType(rootfsFilename)
+		if err != nil {
+			return cli.NewExitError("imgFilesystemType error: "+err.Error(), errArtifactCreate)
+		}
+		switch fstype {
+		case fat:
+			err = runFsck(rootfsFilename, "vfat")
+		case ext:
+			err = runFsck(rootfsFilename, "ext4")
+		case unsupported:
+			err = errors.New("writeRootfs: unsupported filesystem")
+
+		}
+		if err != nil {
+			return cli.NewExitError("runFsck error: "+err.Error(), errArtifactCreate)
+		}
 	}
```

Comment:
I originally considered running `fsck` for all images at `writeRootfs`.

On the one hand many tests would require fixing, but on the other hand I realized it won't be a good idea anyway as it will be changing filesystem checksums. Hence my final conclusion on only "fixing" images generated via SSH.